sdk/core/core-client/src/interfaces.ts

```diff
@@ -0,0 +1,410 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AbortSignalLike } from "@azure/abort-controller";
+import { OperationTracingOptions } from "@azure/core-tracing";
+import {
+  HttpMethods,
+  PipelineResponse,
+  TransferProgressEvent,
+  PipelineRequest
+} from "@azure/core-https";
+
+export interface OperationRequest extends PipelineRequest {
```

Comment:
This was a great comment because it made me realize that extending the request this way didn't work great with `clone()` on the request. I fixed that bug by moving these props into a grab-bag option that `clone()` can persist!